app/src/components/features/rules/RuleBuilder/Header/ActionButtons/CreateRuleButton/index.js

```diff
@@ -129,11 +131,9 @@ const CreateRuleButton = ({
           }
         }
 
-        const ruleId = currentlySelectedRuleData.id;
-
-        if (!isRuleEditorModal) {
-          redirectToRuleEditor(navigate, ruleId, "create");
-        }
+        // It is necessary to redirect to editor after all the states have been updated so
+        // that usePrompt does not block the navigation
```

Comment:
move the comment above the affected `useEffect` 
Also edit it to explain how this is specifically waiting for `isCurrentlySelectedRuleHasUnsavedChanges` state change  and why